packages/next/src/shared/lib/get-img-props.ts

```diff
@@ -585,31 +585,41 @@ export function getImgProps(
     style
   )
 
-  const blurStyle =
-    placeholder === 'blur' && blurDataURL && !blurComplete
-      ? {
-          backgroundSize: imgStyle.objectFit || 'cover',
-          backgroundPosition: imgStyle.objectPosition || '50% 50%',
-          backgroundRepeat: 'no-repeat',
-          backgroundImage: `url("data:image/svg+xml;charset=utf-8,${getImageBlurSvg(
-            {
-              widthInt,
-              heightInt,
-              blurWidth,
-              blurHeight,
-              blurDataURL,
-              objectFit: imgStyle.objectFit,
-            }
-          )}")`,
-        }
-      : {}
+  const backgroundImage =
+    !blurComplete && placeholder !== 'empty'
+      ? placeholder === 'blur' && blurDataURL
+        ? `url("data:image/svg+xml;charset=utf-8,${getImageBlurSvg({
+            widthInt,
+            heightInt,
+            blurWidth,
+            blurHeight,
+            blurDataURL,
+            objectFit: imgStyle.objectFit,
+          })}")`
+        : placeholder === 'blur'
+        ? null
+        : `url("${placeholder}")`
+      : null
+
+  let placeholderStyle = backgroundImage
+    ? {
+        backgroundSize: imgStyle.objectFit || 'cover',
+        backgroundPosition: imgStyle.objectPosition || '50% 50%',
+        backgroundRepeat: 'no-repeat',
+        backgroundImage,
+      }
+    : {}
```

Comment:
I think you can simplify this to make it look more like the original.

Try this:

```ts
  const placeholderStyle =
    placeholder !== 'empty' && !blurComplete
      ? {
          backgroundSize: imgStyle.objectFit || 'cover',
          backgroundPosition: imgStyle.objectPosition || '50% 50%',
          backgroundRepeat: 'no-repeat',
          backgroundImage: placeholder === 'blur' && blurDataURL 
            ? `url("data:image/svg+xml;charset=utf-8,${getImageBlurSvg(
              {
              widthInt,
              heightInt,
              blurWidth,
              blurHeight,
              blurDataURL,
              objectFit: imgStyle.objectFit,
              }
            )}")`
            : `url("${placeholder}")`,
      }
    : {}
```

Reply:
I've decided to refactor it a bit for better type narrowing. With your suggestion the `placeholder` is narrowed to:

<img width="566" alt="Screenshot 2023-08-03 at 18 36 41" src="https://github.com/vercel/next.js/assets/28541613/9626436c-af69-49b2-bcfb-a036851270fe">

With my change it properly narrows it to just `data:image/...`:

<img width="572" alt="Screenshot 2023-08-03 at 18 37 06" src="https://github.com/vercel/next.js/assets/28541613/cfa1f63a-5f80-47ac-89d6-3bd23f7d6a6d">

I know we're throwing error when `placeholder="blur"` is set and `blurDataURL` is not passed, but I think it's still worth to clean things up here